### PR TITLE
Add json data files To OOP Ruby School Library

### PR DESCRIPTION
**Pull request by codeWithHyder**
- Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).

- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)

- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: data/books.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[
+  {
+    "title": "The Atomic Habits",
+    "author": "John",
+    "rentals": [
+
+    ]
+  }
+]
```
Comment (nahnahdev) [inline data/books.json]: - [ ] Lastly, I deleted all the current `json` files in the project to test if the part of the project requirement that mentioned that you need to remember to handle the case when one or more of the files are missing and ensure that the app works and shows no errors 👍 
- [ ] The app did work as expected. However upon exist of the app, the `json` files weren't recreated in the `data directory` where they were previously to store the new data. As a result, data wasn't persisted in your app as expected so kindly fix that as well 😉 
- [ ] If you look on the left part of the screen in the video, you'll notice that on exit of the app the json files weren't created to store data 💯 
- Video Reference
_**I took this video after deleting all the json files from the project 👍**_

https://github.com/codeWithHyder/my-school-library/assets/102699958/42e259bc-8e31-44aa-b3f9-718bcd476592

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @codeWithHyder and @ReemMohamedAbdelfatah , 

Great work so far. There are just a few changes you need to make and you’ll be on your way to the next project.


 ## Highlights 🔥 
Good use of GitFlow ✔️
Linters well setup and passing ✔️
Most project requirements met ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 def main
   app = App.new
   puts 'Welcome to the School Library App!'
-  run_app(app)
+  app.run
 end
```
Comment (Owusu-Desmond) [inline main.rb:7]: - [ ] Please while making the changes requested by the previous reviewer, I am getting an error while running your application so ensure that you fix this :point_down: and make sure your fix the changes required by the previous reviewer :relaxed: 

image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @codeWithHyder :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 2 above)
Comment (codeWithHyder) [inline main.rb:7]: fixed the issue

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @codeWithHyder 

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good use of ruby syntax✔️ 
- Kudus to heed the required changes as stated previously  👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**